```
________________________________ test_write_gif ________________________________

simulation = ===========================
--------------------
Reader performance:
--------------------
/root/project/tests/test_dat...00:00 UTC
        Calculation steps: 5 * 1:00:00 - total time: 5:00:00
        Output steps: 6 * 1:00:00
===========================

tmpdir = local('/tmp/pytest-of-root/pytest-0/test_write_gif0')
benchmark = <pytest_benchmark.fixture.BenchmarkFixture object at 0x7f3581bb1fd0>

    def test_write_gif(simulation, tmpdir, benchmark):
>       benchmark(simulation.animation, filename = tmpdir / 'test.gif')

tests/plotting/test_write_animation.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/conda/envs/opendrift/lib/python3.14/site-packages/pytest_benchmark/fixture.py:125: in __call__
    return self._raw(function_to_benchmark, *args, **kwargs)
/opt/conda/envs/opendrift/lib/python3.14/site-packages/pytest_benchmark/fixture.py:171: in _raw
    function_result = function_to_benchmark(*args, **kwargs)
opendrift/models/basemodel/__init__.py:3163: in animation
    fig.canvas.draw()
/opt/conda/envs/opendrift/lib/python3.14/site-packages/matplotlib/backends/backend_agg.py:438: in draw
    self.figure.draw(self.renderer)
/opt/conda/envs/opendrift/lib/python3.14/site-packages/matplotlib/artist.py:94: in draw_wrapper
    result = draw(artist, renderer, *args, **kwargs)
/opt/conda/envs/opendrift/lib/python3.14/site-packages/matplotlib/artist.py:71: in draw_wrapper
    return draw(artist, renderer)
/opt/conda/envs/opendrift/lib/python3.14/site-packages/matplotlib/figure.py:3276: in draw
    self.get_layout_engine().execute(self)
/opt/conda/envs/opendrift/lib/python3.14/site-packages/matplotlib/layout_engine.py:188: in execute
    kwargs = get_tight_layout_figure(
/opt/conda/envs/opendrift/lib/python3.14/site-packages/matplotlib/_tight_layout.py:266: in get_tight_layout_figure
    kwargs = _auto_adjust_subplotpars(fig, renderer,
/opt/conda/envs/opendrift/lib/python3.14/site-packages/matplotlib/_tight_layout.py:82: in _auto_adjust_subplotpars
    bb += [martist._get_tightbbox_for_layout_only(ax, renderer)]
/opt/conda/envs/opendrift/lib/python3.14/site-packages/matplotlib/artist.py:1485: in _get_tightbbox_for_layout_only
    return obj.get_tightbbox(*args, **{**kwargs, "for_layout_only": True})
/opt/conda/envs/opendrift/lib/python3.14/site-packages/cartopy/mpl/geoaxes.py:483: in get_tightbbox
    return super().get_tightbbox(renderer, *args, **kwargs)
/opt/conda/envs/opendrift/lib/python3.14/site-packages/matplotlib/axes/_base.py:4763: in get_tightbbox
    bbox = a.get_tightbbox(renderer)
/opt/conda/envs/opendrift/lib/python3.14/site-packages/cartopy/mpl/gridliner.py:1251: in get_tightbbox
    return mtrans.Bbox.union(bboxes)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

bboxes = [None, None]

    @staticmethod
    def union(bboxes):
        """Return a `Bbox` that contains all of the given *bboxes*."""
        if not len(bboxes):
            raise ValueError("'bboxes' cannot be empty")
>       x0 = np.min([bbox.xmin for bbox in bboxes])
E       AttributeError: 'NoneType' object has no attribute 'xmin'

/opt/conda/envs/opendrift/lib/python3.14/site-packages/matplotlib/transforms.py:670: AttributeError
```

I'm not getting this error locally, probably only on newer packages. Will have to check later.

--- a/opendrift/models/basemodel/__init__.py
+++ b/opendrift/models/basemodel/__init__.py
@@ -2626,6 +2626,17 @@ class OpenDriftSimulation(PhysicsMethods, Timeable, Configurable):
                 pass
         gl._draw_gridliner = _safe_draw_gridliner
 
+        # Cartopy's gridliner.get_tightbbox can raise AttributeError when label
+        # artists have no bbox (e.g. after _draw_gridliner silently failed).
+        # Returning None is valid for matplotlib – it means "no visible bbox".
+        _original_get_tightbbox = gl.get_tightbbox
+        def _safe_get_tightbbox(*args, **kwargs):
+            try:
+                return _original_get_tightbbox(*args, **kwargs)
+            except (AttributeError, ValueError):
+                return None
+        gl.get_tightbbox = _safe_get_tightbbox
+
         if 'ocean_color' in kwargs:
             ax.patch.set_facecolor(kwargs['ocean_color'])
             ocean_color = kwargs['ocean_color']

--- a/opendrift/models/basemodel/__init__.py
+++ b/opendrift/models/basemodel/__init__.py
@@ -2612,30 +2612,7 @@ class OpenDriftSimulation(PhysicsMethods, Timeable, Configurable):
             lonmax -= .1  # To avoid problem with Cartopy
         ax.set_extent([lonmin, lonmax, latmin, latmax], crs=self.crs_lonlat)
 
-        gl = ax.gridlines(self.crs_lonlat, draw_labels=True, xlocs=xlocs, ylocs=ylocs)
-        gl.top_labels = None
-
-        # Cartopy's _draw_gridliner can fail with Shapely 2.x when the map
-        # boundary path is not a closed ring.  Wrap it so rendering never
-        # crashes, even on older Cartopy builds.
-        _original_draw_gridliner = gl._draw_gridliner
-        def _safe_draw_gridliner(*args, **kwargs):
-            try:
-                return _original_draw_gridliner(*args, **kwargs)
-            except Exception:
-                pass
-        gl._draw_gridliner = _safe_draw_gridliner
-
-        # Cartopy's gridliner.get_tightbbox can raise AttributeError when label
-        # artists have no bbox (e.g. after _draw_gridliner silently failed).
-        # Returning None is valid for matplotlib – it means "no visible bbox".
-        _original_get_tightbbox = gl.get_tightbbox
-        def _safe_get_tightbbox(*args, **kwargs):
-            try:
-                return _original_get_tightbbox(*args, **kwargs)
-            except (AttributeError, ValueError):
-                return None
-        gl.get_tightbbox = _safe_get_tightbbox
+        gl = ax.gridlines(self.crs_lonlat, draw_labels=['left', 'bottom'], xlocs=xlocs, ylocs=ylocs)
 
         if 'ocean_color' in kwargs:
             ax.patch.set_facecolor(kwargs['ocean_color'])

--- a/opendrift/models/basemodel/__init__.py
+++ b/opendrift/models/basemodel/__init__.py
@@ -336,7 +336,7 @@ class OpenDriftSimulation(PhysicsMethods, Timeable, Configurable):
         # Set up logging
         logformat = '%(asctime)s %(levelname)-7s %(name)s:%(lineno)d: %(message)s'
         datefmt = '%H:%M:%S'
-        
+
         if loglevel < 10:  # 0 is NOTSET, giving no output
             print('WARNING: from next version (1.14.10), loglevel of 0 will give no logging, please change to 10 for DEBUG')
             loglevel = 10
@@ -1676,16 +1676,16 @@ class OpenDriftSimulation(PhysicsMethods, Timeable, Configurable):
         elements = self.ElementType(lon=lon, lat=lat, z=-z)
 
         self.schedule_elements(elements, time)
-    
+
     @require_mode(mode=Mode.Ready)
     def seed_from_dataset(self, ds, trajectory_time_index=-1, time=None, keep_properties=True, **kwargs):
         """Seed elements from OpenDrift dataset
 
          Arguments:
-            ds                      (DataArray)         :   DataArray from previous OpenDrift run. 
-            trajectory_time_index   (int)               :   Time index from which to continue OpenDrift run. 
-            time                    (datenum or list)   :   Time to initiate particles. If None, uses time from trajectory_time_index. 
-            keep_properties         (bool)              :   Keep element properties from DataArray. If False, overrides properties with new ones. 
+            ds                      (DataArray)         :   DataArray from previous OpenDrift run.
+            trajectory_time_index   (int)               :   Time index from which to continue OpenDrift run.
+            time                    (datenum or list)   :   Time to initiate particles. If None, uses time from trajectory_time_index.
+            keep_properties         (bool)              :   Keep element properties from DataArray. If False, overrides properties with new ones.
         """
         ds = ds.isel(time=trajectory_time_index)
 
@@ -1695,9 +1695,9 @@ class OpenDriftSimulation(PhysicsMethods, Timeable, Configurable):
 
         # Dropping trajectories which had not been initiated at selected time, e.g. for continuous release.
         ds = ds.where(ds.age_seconds >= 0, drop=True)
-        
+
         logger.info('Using positions from dataset at time %s' % (str(time)))
-        
+
         try:
             file_class = ds.opendrift_class
             current_class = self.__class__.__name__
@@ -1718,12 +1718,12 @@ class OpenDriftSimulation(PhysicsMethods, Timeable, Configurable):
                 else:
                     if key in ds:
                         prop_dict[key] = ds[key].values
-            
+
             logger.info('Seeding %i particles from dataset' %(len(ds.lon)))
             logger.info('Using values from dataset for element properties: ')
             logger.info('%s' % (str([key for key in prop_dict.keys()])))
             self.seed_elements(ds.lon, ds.lat, time=time, **prop_dict)
-        
+
         else:
             logger.info('Using only lon, lat from provided dataset. Omitting particle properties from previous run')
             self.seed_elements(ds.lon, ds.lat, time=time, **kwargs)
@@ -1732,12 +1732,12 @@ class OpenDriftSimulation(PhysicsMethods, Timeable, Configurable):
     @require_mode(mode=Mode.Ready)
     def seed_from_file(self, filename, trajectory_time_index=-1, time=None, keep_properties=True, **kwargs):
         """Seed elements from OpenDrift output netCDF file
-        
+
         Arguments:
             filename                (str)               :   Name of netCDF file with particle positions.
-            trajectory_time_index   (int)               :   Time index from which to continue OpenDrift run. 
-            time                    (datenum or list)   :   Time to initiate particles. If None, uses time from trajectory_time_index. 
-            keep_properties         (bool)              :   Keep element properties from file. If False, overrides properties with new ones. 
+            trajectory_time_index   (int)               :   Time index from which to continue OpenDrift run.
+            time                    (datenum or list)   :   Time to initiate particles. If None, uses time from trajectory_time_index.
+            keep_properties         (bool)              :   Keep element properties from file. If False, overrides properties with new ones.
         """
         logger.info('Seeding elements from previous run in %s' %(filename))
         ds = xr.open_dataset(filename)
@@ -2614,6 +2614,17 @@ class OpenDriftSimulation(PhysicsMethods, Timeable, Configurable):
 
         gl = ax.gridlines(self.crs_lonlat, draw_labels=True, xlocs=xlocs, ylocs=ylocs)
         gl.top_labels = None
+
+        # Cartopy's _draw_gridliner can fail with Shapely 2.x when the map
+        # boundary path is not a closed ring.  Wrap it so rendering never
+        # crashes, even on older Cartopy builds.
+        _original_draw_gridliner = gl._draw_gridliner
+        def _safe_draw_gridliner(*args, **kwargs):
+            try:
+                return _original_draw_gridliner(*args, **kwargs)
+            except Exception:
+                pass
+        gl._draw_gridliner = _safe_draw_gridliner
 
         if 'ocean_color' in kwargs:
             ax.patch.set_facecolor(kwargs['ocean_color'])

